.githooks/pre-receive

```diff
@@ -1,28 +1,7 @@
 #!/usr/bin/env bash
 
-# An example hook script to verify what is about to be pushed.  Called by "git
-# push" after it has checked the remote status, but before anything has been
-# pushed.  If this script exits with a non-zero status nothing will be pushed.
-#
-# This hook is called with the following parameters:
-#
-# $1 -- Name of the remote to which the push is being done
-# $2 -- URL to which the push is being done
-#
-# If pushing without using a named remote those arguments will be equal.
-#
-# Information about the commits which are being pushed is supplied as lines to
-# the standard input in the form:
-#
-#   <local ref> <local oid> <remote ref> <remote oid>
-#
-# This sample shows how to prevent push of commits where the log message starts
-# with "WIP" (work in progress).
-
-remote="$1"
-url="$2"
-
-BRANCH_CHANGED_FILES=($(git diff --name-only main..HEAD))
+# Usage: bash pre-receive <CHANGED FILES>
+BRANCH_CHANGED_FILES=($1)
```

Comment:
I think this might be more correct (?):

```suggestion
BRANCH_CHANGED_FILES=($@)
```

Reply:
Haha, nice timing @h5law! I just edited my comment - I'm on my way out of the house and had to step away for a sec.

Reply:
When we pass `"${{needs.changedfiles.outputs.all}}"` what is actually happening is the list of filenames is expanded into this form `bash pre-recieve file1 file2 file3 file4 ...` so when we do `($@)` bash is actually taking all these individual file names and storing them into an array we can loop over. 👍🏼 

I think it would be worth looking into the reason why the workflow would not find head as the previous solution was cleaner IMO and would remove the need for the `changedfiles` workflow aswell.